controllers/resource/resource_generator.go

```diff
@@ -196,6 +203,19 @@ func updateCustomVolumes(desiredDeployment, patchingDeployment *appsv1.Deploymen
 	}
 }
 
+func updateCustomStartupProbe(desiredDeployment, patchingDeployment *appsv1.Deployment) {
+	for _, containerGenerated := range desiredDeployment.Spec.Template.Spec.Containers {
+		for _, containerProvided := range patchingDeployment.Spec.Template.Spec.Containers {
+			if containerGenerated.Name == containerProvided.Name && containerProvided.StartupProbe != nil {
+				log.Log.Info("Found custom startup probe in custom deployment. Injecting into the deployment generated by the operator.")
+				if containerProvided.StartupProbe.FailureThreshold != 0 {
```

Comment:
Because 0 means the Threshold has not been set.
Although I have to admit, I'm not entirely sure what a value of zero would do here. Does it mean endless?

Reply:
As discussed, we can leave it as is.